modules/EED_SquareOnsiteOAuth.module.php

```diff
@@ -777,4 +834,21 @@ public static function closeOauthWindow($msg = null)
         echo $js_out;
         die();
     }
+
+
+    /**
+     * Close the OAuth window with JS.
+     *
+     * @param string $err_msg
+     * @param bool   $alert
+     * @return void
+     */
+    public static function returnJsonError(string $err_msg = '', bool $alert = false)
```

Comment:
![ACKCHYUALLY](https://user-images.githubusercontent.com/1751030/145262771-91b7207e-c76c-42dd-b76b-209f96dccd65.jpg)

technically this doesn't "return" the error, so maybe:

```suggestion
    public static function echoJsonError(string $err_msg = '', bool $alert = false)
```